pkg/credentialprovider/gcp/metadata.go

```diff
@@ -148,18 +166,73 @@ func (g *dockerConfigUrlKeyProvider) Provide() credentialprovider.DockerConfig {
 	return credentialprovider.DockerConfig{}
 }
 
+func runWithBackoff(url string, f func() ([]byte, error)) ([]byte, error) {
```

Comment:
The other option is for this to cause a panic. Not sure if that's better or not, so this way lgtm.